backend/lib/endpoints/posts.js

```diff
@@ -418,41 +504,58 @@ async function routes(app) {
     "/:postId/comments/:commentId",
     { preValidation: [app.authenticate], schema: deleteCommentSchema },
     async (req) => {
-      const { params, userId } = req;
-      const { commentId, postId } = params;
-      const { ok, deletedCount } = await Comment.deleteMany({
-        $or: [
-          { _id: commentId, authorId: userId, postId },
-          { parentId: commentId, postId },
-        ],
-      });
-      if (ok !== 1 || deletedCount < 1) {
-        return new httpErrors.BadRequest();
+      const { userId } = req.body;
+      const { commentId } = req.params;
+      const [findErr, comment] = await app.to(Comment.findById(commentId));
+      if (findErr) {
+        throw app.httpErrors.notFound();
+      } else if (comment.author.id !== userId) {
+        throw app.httpErrors.forbidden();
+      }
+
+      const [deleteCommentErr, deletedComment] = await app.to(comment.delete());
+      if (deleteCommentErr) {
+        req.log.error("Failed deleting comment", { deleteCommentErr });
+        throw app.httpErrors.internalServerError();
       }
-      return { deletedCount, success: true };
+
+      const {
+        deletedCount: deletedNestedCount,
+        ok: deleteNestedOk,
+      } = await Comment.deleteMany({ parentId: commentId });
+      if (deleteNestedOk !== 1) {
+        app.log.error("failed removing nested comments for deleted comment", {
+          commentId,
+        });
+      }
+
+      return {
+        deletedComment,
+        deletedCount: deletedNestedCount + 1,
+        success: true,
+      };
     },
   );
 
   app.put(
     "/:postId/comments/:commentId/likes/:userId",
     { preValidation: [app.authenticate], schema: likeUnlikeCommentSchema },
     async (req) => {
-      const { commentId, postId, userId } = req.params;
-      if (userId !== req.userId) {
-        return new httpErrors.Forbidden();
-      }
-      const updatedComment = await Comment.findOneAndUpdate(
-        { _id: commentId, likes: { $ne: userId }, postId },
-        { $inc: { likesCount: 1 }, $push: { likes: userId } },
-        { new: true },
+      const { commentId, userId } = req.params;
+      const [updateErr, updatedComment] = await app.to(
```

Comment:
first this needs to check if `req.params.userId === req.userId` or you could add someone else's like